src/models/PenaltyIdentifierSchemaFactory.ts

```diff
@@ -0,0 +1,94 @@
+import Joi from '@hapi/joi';
+
+type PrefixVariation = { upperCase: string[]; lowerCase: string[]; };
+type PrefixMap = { singleCharacter: PrefixVariation, doubleCharacter: PrefixVariation; };
+
+export class PenaltyIdentifierSchemaFactory {
+
+    public readonly companyNumberRegex: RegExp;
+
+    public readonly penaltyReferenceRegex: RegExp;
+
+    constructor(companyNumberPrefixes: string) {
+
+        if (companyNumberPrefixes.length === 0) {
+            throw new Error('Company prefixes must not be empty. e.g. NI,SC,SO');
+        }
+
+        if (companyNumberPrefixes === '*') {
```

Comment:
What is meant by this? Isnt the list the one of the config? when would we have * as an item

Reply:
Making it configurable includes having a way to specify a way to match all company prefixes. `*` will match any company number between 1 and 8 chars long using  alphanumeric characters

Reply:
the ticket only mentions allowing company numbers with the prefix of SC, NI, OC, SO, R, so what would be the reason behind allowing anything outside of what is specified in this configuration?

Reply:
ah its the default value when the list is empty? Shouldn't we just throw an error? Seems like a rather important configuration

Reply:
I'll ask Bart to clarify that because the scenario 2 of my ticket is about making the prefixes configurable which in interpretation can mean including allowing any. Also, if for some reason the business decided to allow any prefix.

Reply:
![image](https://user-images.githubusercontent.com/54070318/85120712-8bf26080-b21b-11ea-9776-2b7035a25961.png)

Reply:
![image](https://user-images.githubusercontent.com/54070318/85121114-22bf1d00-b21c-11ea-84c7-0224baced2c9.png)